filters/encrypt/filter.go

```diff
@@ -0,0 +1,542 @@
+			return nil, fmt.Errorf("%s: %w", op, err)
+		}
+	case pKind == reflect.Slice:
+		switch {
+		// if the field is a slice of string or slice of []byte
+		case pType == reflect.TypeOf([]string{}) || pType == reflect.TypeOf([]*string{}) || pType == reflect.TypeOf([][]uint8{}):
+			ef.l.RLock()
+			classificationTag := getClassificationFromTagString(string(SecretClassification), withFilterOperations(ef.FilterOperationOverrides))
+			ef.l.RUnlock()
+			if err := ef.filterSlice(ctx, classificationTag, payloadValue, opts...); err != nil {
+				return nil, fmt.Errorf("%s: %w", op, err)
+			}
+		// if the field is a slice of structs, recurse through them...
+		default:
+			for i := 0; i < payloadValue.Len(); i++ {
+				f := payloadValue.Index(i)
+				if f.Kind() == reflect.Ptr {
+					f = f.Elem()
+				}
+				if f.Kind() != reflect.Struct {
+					continue
+				}
+				if err := ef.filterField(ctx, f, opts...); err != nil {
+					return nil, fmt.Errorf("%s: %w", op, err)
+				}
+			}
+		}
+	case pKind == reflect.Struct:
+		if err := ef.filterField(ctx, payloadValue, opts...); err != nil {
+			return nil, fmt.Errorf("%s: %w", op, err)
+		}
+	}
+	return e, nil
+}
+
+// filterField will recursively iterate over all the fields for a struct value
+// and filter them based on their DataClassification
+func (ef *Filter) filterField(ctx context.Context, v reflect.Value, opt ...Option) error {
+	const op = "event.(Filter).filterField"
+	// check for nil value (prevent panics)
+	if v == reflect.ValueOf(nil) {
+		return nil
+	}
+
+	for i := 0; i < v.Type().NumField(); i++ {
+		field := v.Field(i)
+
+		switch v.Field(i).Kind() {
+		case reflect.Ptr, reflect.Interface:
+			field = v.Field(i).Elem()
+			if field == reflect.ValueOf(nil) {
+				continue
+			}
+		}
+
+		fkind := field.Kind()
+		ftype := field.Type()
+
+		var taggedInterface Taggable
+		var isTaggable bool
+		// check exported fields to see if they implement the Taggable interface
+		if field.CanSet() {
+			taggedInterface, isTaggable = v.Field(i).Interface().(Taggable)
+		}
+
+		switch {
+		// if the field is a string or []byte then we just need to sanitize it
+		case ftype == reflect.TypeOf("") || ftype == reflect.TypeOf([]uint8{}):
+			ef.l.RLock() // passing a ref to the FilterOperationOverrides map
+			classificationTag := getClassificationFromTag(v.Type().Field(i).Tag, withFilterOperations(ef.FilterOperationOverrides))
+			ef.l.RUnlock()
+			if err := ef.filterValue(ctx, field, classificationTag, opt...); err != nil {
+				return fmt.Errorf("%s: %w", op, err)
+			}
+		// if the field is a slice
+		case fkind == reflect.Slice:
+			switch {
+			// if the field is a slice of string or slice of []byte
+			case ftype == reflect.TypeOf([]string{}) || ftype == reflect.TypeOf([][]uint8{}):
+				ef.l.RLock() // passing a ref to the FilterOperationOverrides map
+				classificationTag := getClassificationFromTag(v.Type().Field(i).Tag, withFilterOperations(ef.FilterOperationOverrides))
```

Comment:
It may not be clear to people from the docs that the order of the values in the tag string matters. It might be good to call that out, or to modify the function so that it doesn't actually matter.

Reply:
Add the tag fields to the README. ty!